test/terraform_aws_ecs_service_test.go

```diff
@@ -30,3 +42,129 @@ func TestTerraformAwsEcsServiceSimple(t *testing.T) {
 	terraform.InitAndApply(t, terraformOptions)
 
 }
+
+// Retrieve tasks associated with a cluster
+func GetTasks(t *testing.T, region string, clusterName string) *ecs.ListTasksOutput {
+	taskList, err := GetTasksE(t, region, clusterName)
+	require.NoError(t, err)
+	return taskList
+}
+
+func GetTasksE(t *testing.T, region string, clusterName string) (*ecs.ListTasksOutput, error) {
+	var returnTaskList *ecs.ListTasksOutput
+	ecsClient, err := aws.NewEcsClientE(t, region)
+	if err != nil {
+		return returnTaskList, err
+	}
+
+	params := &ecs.ListTasksInput{
+		Cluster: awssdk.String(clusterName),
+	}
+
+	// Need to spin and wait to allow time for resources to get up
+	maxRetries := 3
+	retryDuration, _ := time.ParseDuration("30s")
+	_, err = retry.DoWithRetryE(t, "Get tasks", maxRetries, retryDuration,
+		func() (string, error) {
+			tasks, _ := ecsClient.ListTasks(params)
+
+			if len(tasks.TaskArns) == 0 {
+				return "Did not retrieve tasks", fmt.Errorf("We returned empty tasks %v", tasks.TaskArns)
+			}
+			returnTaskList = tasks
+			return "Retrieved tasks", nil
+		},
+	)
+	if err != nil {
+		return returnTaskList, err
+	}
+
+	return returnTaskList, nil
+}
+
+// Retrieve ENI from task ARNs and cluster
+func GetEni(t *testing.T, region string, cluster string, taskArns []*string) *string {
+	task, err := GetEniE(t, region, cluster, taskArns)
+	require.NoError(t, err)
+	return task
```

Comment:
Feels odd to return `task` when the intent of a function called `GetEni` is presumably to return the Eni. Why not return Eni? Given `GetEniE` below actually returns the Eni, assuming `eniDetail := returnedTasks.Tasks[0].Attachments[0].Details[1].Value` is the Eni.